modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradeJavaBaseModelListenerCheck.java

```diff
@@ -0,0 +1,142 @@
+import com.liferay.source.formatter.parser.JavaClassParser;
+import com.liferay.source.formatter.parser.JavaMethod;
+import com.liferay.source.formatter.parser.JavaParameter;
+import com.liferay.source.formatter.parser.JavaSignature;
+import com.liferay.source.formatter.parser.JavaTerm;
+
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.apache.commons.lang.StringUtils;
+
+/**
+ * @author Michael Cavalcanti
+ */
+public class UpgradeJavaBaseModelListenerCheck extends BaseUpgradeCheck {
+
+	@Override
+	protected String format(
+			String fileName, String absolutePath, String content)
+		throws Exception {
+
+		JavaClass javaClass = JavaClassParser.parseJavaClass(fileName, content);
+
+		List<String> implementedClassNames =
+			javaClass.getImplementedClassNames();
+
+		if (implementedClassNames.contains("BaseModelListener")) {
+			return content;
+		}
+
+		for (JavaTerm childJavaTerm : javaClass.getChildJavaTerms()) {
+			if (!childJavaTerm.isJavaMethod()) {
+				continue;
+			}
+
+			JavaMethod javaMethod = (JavaMethod)childJavaTerm;
+
+			String javaMethodContent = javaMethod.getContent();
+
+			String newJavaMethodContent = _checkMethodDefinition(
+				javaMethodContent, javaMethod, _onAfterUpdatePattern);
+
+			newJavaMethodContent = _checkMethodDefinition(
+				newJavaMethodContent, javaMethod, _onfBeforeUpdatePattern);
+
+			content = StringUtil.replace(
+				content, javaMethodContent, newJavaMethodContent);
+		}
+
+		return content;
+	}
+
+	private String _checkMethodDefinition(
+		String javaMethodContent, JavaMethod javaMethod, Pattern pattern) {
+
+		Matcher matcher = pattern.matcher(javaMethodContent);
+
+		if (matcher.find()) {
+			Matcher superMatcher = _superPattern.matcher(javaMethodContent);
+
+			if (superMatcher.find()) {
+				javaMethodContent = _checkSuper(
+					javaMethodContent, superMatcher.start());
+			}
+
+			JavaSignature javaSignature = javaMethod.getSignature();
+
+			List<JavaParameter> parameters = javaSignature.getParameters();
+
+			if (parameters.size() != 1) {
+				return javaMethodContent;
+			}
+
+			JavaParameter javaParameter = parameters.get(0);
+
+			String javaParameterName = javaParameter.getParameterName();
+
+			String javaParameterType = javaParameter.getParameterType();
+
+			String newParameter = StringBundler.concat(
+				javaParameterType, " original",
+				StringUtils.capitalize(javaParameterName));
+
+			String newParameters = StringBundler.concat(
+				newParameter, StringPool.COMMA_AND_SPACE, javaParameterType,
+				StringPool.SPACE, javaParameterName);
+
+			String methodCall = JavaSourceUtil.getMethodCall(
+				javaMethodContent, matcher.start());
+
+			javaMethodContent = StringUtil.replace(
+				javaMethodContent, JavaSourceUtil.getParameters(methodCall),
+				newParameters);
+		}
+
+		return javaMethodContent;
+	}
+
+	private String _checkSuper(String javaMethodContent, int pos) {
```

Comment:
@michaelmdelcavalcanti, to ensure consistency we should rename it to `_formatSuper`